egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -597,8 +617,9 @@ while [ $x -lt $num_iters ]; do
         $cmd $train_queue_opt $dir/log/train.$x.$n.log \
           nnet3-train $parallel_train_opts --print-interval=10 --momentum=$momentum \
           --max-param-change=$max_param_change \
-          --optimization.min-deriv-time=$min_deriv_time "$raw" \
-          "ark:nnet3-copy-egs $context_opts ark:$cur_egs_dir/egs.$archive.ark ark:- | nnet3-shuffle-egs --buffer-size=$shuffle_buffer_size --srand=$x ark:- ark:-| nnet3-merge-egs --minibatch-size=$this_num_chunk_per_minibatch --measure-output-frames=false --discard-partial-minibatches=true ark:- ark:- |" \
+          --optimization.min-deriv-time=$min_deriv_time \
+          --num-splits-per-chunk=$[$chunk_width/$minibatch_chunk_size] "$raw" \
+          "ark:nnet3-copy-egs $context_opts ark:$cur_egs_dir/egs.$archive.ark ark:- | nnet3-shuffle-egs --buffer-size=$shuffle_buffer_size --srand=$x ark:- ark:-| nnet3-merge-egs --minibatch-size=$this_num_chunk_per_minibatch --measure-output-frames=false --discard-partial-minibatches=false ark:- ark:- | nnet3-split-egs --chunk-size-after-split=$minibatch_chunk_size \"$raw\" ark:- ark:- | nnet3-add-recurrent-io-to-egs \"$raw\" ark:- ark:- |" \
```

Comment:
I was thinking that if the chunk size is large. discarding the partial minibatch after merge-egs may lose data i quantity that is not negligible.  For example, if chunk size=20, then there are 20000 egs per ark, and we will at most discard 100 out of 20000 egs. But if chunk size=200, then there are 2000 egs per ark, and we will at most discard 100 out of 2000 egs. (assume minibatch size=100)